providers/mlx5/mlx5dv_dr.h

```diff
@@ -1042,6 +1047,7 @@ struct dr_icm_buddy_mem {
 
 	/* This is the list of used chunks. HW may be accessing this memory */
 	struct list_head	used_list;
+	unsigned int		used_memory;
```

Comment:
Memory is size_t

Reply:
This value is per body which can reach 64M bytes and not exposed outside from mlx5, 32 bits is much larger than is really needed.

Reply:
The size_t usage is about correctness and not about "reach".

Reply:
Updated accordingly.